### PR TITLE
 feat(pipeline_template): Better support for templated pipelines with dynamic sources - Take 2

### PR DESCRIPTION
**Pull request by jervi**
This is the second iteration of #4288, which had to be rolled back in #4362.

It was an oversight by me, I was not really aware that the pipeline executions sometimes had both a build number and a pipeline parent. This is fixed.

Original PR:
* Add "Rendered pipeline" tab to "Edit as JSON" modal for templated pipelines
* Don't display error message when editing templated pipelines with dynamic source - real fix in Orca
* Render execution graph when editing templated pipelines with dynamic source - real fix in Orca
* Display warning in Deck if configuration can't be rendered because no executions have been run
* Display warning in Deck if a pipeline with dynamic source can not be rendered
* Display information about which execution used for rendering
* Let the user select which configuration that's used for rendering

### CODE REVIEW

File: app/scripts/modules/core/src/pipeline/config/actions/json/editPipelineJsonModal.html [hunk 1]
```diff
@@ -3,7 +3,12 @@
   <div class="modal-header">
     <h3><span ng-if="!command.locked">Edit </span>{{$ctrl.isStrategy === true ? 'Strategy' : 'Pipeline'}} JSON</h3>
   </div>
-  <div class="modal-body flex-fill">
+  <!-- TODO: remove inline styling below when tabs-basic class has been updated to include list-style: none -->
+  <ul class="tabs-basic" style="list-style: none" ng-if="$ctrl.command.pipelinePlanJSON">
+    <li role="presentation" ng-class="{selected: $ctrl.mode === 'pipeline'}"><a ng-click="$ctrl.mode = 'pipeline'">Configuration</a></li>
+    <li role="presentation" ng-class="{selected: $ctrl.mode === 'renderedPipeline'}"><a ng-click="$ctrl.mode = 'renderedPipeline'">Rendered pipeline</a></li>
+  </ul>
+  <div class="modal-body flex-fill" ng-if="$ctrl.mode == 'pipeline'">
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/actions/json/editPipelineJsonModal.html]: use `===`

Comment (jervi) [reply app/scripts/modules/core/src/pipeline/config/actions/json/editPipelineJsonModal.html]: Fixed 👍

File: app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.html [hunk 2]
```diff
@@ -35,7 +35,32 @@
...
+          </li>
+        </ul>
+      </span>
+      <span ng-if="pipelineExecutions.length === 1"><execution-build-title execution="currentExecution" default-to-timestamp="true" ></execution-build-title></span>
+    </div>
+    <div class="band band-warning" ng-if="pipelineExecutions && !pipelineExecutions.length">
+      <span class="glyphicon glyphicon small glyphicon-alert"></span> This template has a dynamic source. The
+      configuration cannot be rendered before the pipeline has executed at least once.
+    </div>
+    <div class="band band-warning" ng-if="templateError && pipelineExecutions.length && !pipeline.isNew">
```
Comment (danielpeach) [inline app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.html]: This still swallows errors if you're not using dynamic sources. I have a config that doesn't render a pipeline properly - with these changes, when I load the pipeline config screen, there's no indication that anything's wrong.

The edit config modal should open if there's a problem rendering the template - this behavior isn't perfect, but it gives the user a chance to update their config (e.g., in the case they're missing a variable), and then view the rendering error if it still exists.

Comment (jervi) [reply app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.html]: Better?

**Pull request merged**